lib/fathead/mdn_css/fetch.pl

```diff
@@ -17,114 +17,131 @@
-                }
-            );
-        }
-    }
-}
-elsif ( my $error = $tx->error ) {
-    croak sprintf "Error: %d %s while fetching %s", $error->{code} || 0,
-      $error->{message},
-      $tx->req->url;
-}
-
 #downloaded file names will be named 1.html, 2.html ....
-my $file_number = 1;
-
-my $current_active_connections = 0;
-my $maximum_active_connections = 4;
+my $file_number = 1,
+   $current_active_connections = 0,
+   $maximum_active_connections = 4;
 
 my @keyword_urls;
 
 =begin
     save the urls with fragments to a text file called
     fragments.txt so that parse.pl can use this information
     to extract extra information about the fragments
+    used later on in download()
 =cut
 
 open(
-    my $fragments_fh,   '>:encoding(UTF-8)',
-    catfile 'download', 'fragments.txt'
-) or die $!;
-
-for my $url ( keys %urls ) {
-    $url = Mojo::URL->new($url);
-    if ( $url->fragment ) {
+        my $fragments_fh,   '>:encoding(UTF-8)',
+        catfile 'download', 'fragments.txt'
+    ) or die $!;
+
+
+for my $url (@urls_to_fetch) {
+    my $reference_url = Mojo::URL->new($url);
+    my $tx = $ua->get($reference_url);
+    print "fetching url $url\n";
+    fetch($tx);
+    print "downloading $url\n";
+    download();
+}
 
 =begin
-    For fragment in urls past this path /transform-function/ we can get
-    the link for the actual fragment in the following way.
-    Take for example this fragment #matrix()
-    In order to get the actual link, take the fragment, matrix(), remove
-    the brackets, and append it to the original path so that we get the link
-    https://developer.mozilla.org/en-US/docs/Web/CSS/transform-function/matrix
+      We extract the collection of links to keywords from the DOM. The links wanted
+      are found from this part of the DOM:
+      <div class="index">
+        <span>A</span>
+        <ul>
+          <li>
+            <a href="/en-US/docs/Web/CSS/:active"><code>:active</code></a>
+          </li>
+        </ul>
+        ...
+      </div>
 =cut
 
-        if ( $url =~ qr/transform-function/ ) {
-            my $clone =
-              Mojo::URL->new( sprintf "%s://%s", $url->protocol, $url->host );
-
-            #trailing / needed at the end so that when we add a
-            #fragment at the end it does not replace the former last
-            #part of path
-            $clone->path( $url->path . '/' );
-            my $fragment = $url->fragment;
-            $fragment =~ s/\(\)//g;
-            $clone->path($fragment);
-            push @keyword_urls, $clone;
+sub fetch {
+    my $tx = shift;
+
+    if ( $tx->success ) {
+      my $divs = $tx->res->dom->find('div.index, div.column-half');
+        for my $div ( $divs->each ) {
+            for my $ul ( $div->find('ul')->each ) {
+                $ul->find('li')->map(
+                    sub {
+                        my $li = shift;
+                        print "\n value of li $li";
+                        if($li->at('a')) {
```

Comment:
@rasikapohankar Yes, you were on the right track here! I [noticed it too](https://github.com/duckduckgo/zeroclickinfo-fathead/pull/371/files#diff-ca443e93fd9afead3b245f141e784161R41)